community/kernel/src/main/java/org/neo4j/kernel/impl/api/LookupFilter.java

```diff
@@ -45,13 +45,40 @@
      * used by the consistency checker
      */
     public static PrimitiveLongIterator exactIndexMatches( PropertyAccessor accessor,
-            PrimitiveLongIterator indexedNodeIds, int propertyKeyId, Object value )
+            PrimitiveLongIterator indexedNodeIds, IndexQuery... predicates )
     {
-        if ( isNumberOrArray( value ) )
+        if ( !indexedNodeIds.hasNext() )
+        {
+            return indexedNodeIds;
+        }
+
+        IndexQuery[] numericPredicates =
+                Arrays.stream( predicates )
+                        .filter( LookupFilter::isNumericPredicate )
+                        .toArray( IndexQuery[]::new );
+
+        if ( numericPredicates.length > 0 )
         {
-            return PrimitiveLongCollections.filter( indexedNodeIds,
-                    new LookupBasedExactMatchPredicate( accessor, propertyKeyId,
-                            value ) );
+            LongPredicate combinedPredicate = nodeId ->
```

Comment:
I tried to solve this with `Predicates.all` as you suggest, but I couldn't make it better. This is because we need to fetch the property value for every predicate which means that we would need to map them and create lambdas or something.